src/commonMain/kotlin/org/mongodb/kbson/serialization/Serializers.kt

```diff
@@ -68,11 +74,168 @@ import org.mongodb.kbson.BsonTimestamp
+    val encoder = PrimitiveBsonEncoder(serializersModule) { result = it }
+    encoder.encodeSerializableValue(serializer, value)
+    return result
+}
+
+@ExperimentalApi
+internal fun <T> Ejson.readBson(element: BsonValue, deserializer: DeserializationStrategy<T>): T =
+    BsonDecoder(element, serializersModule, ignoreUnknownKeys).decodeSerializableValue(deserializer)
+
+
+/**
+ * Serializes the given [value] into an equivalent [JsonElement] using a serializer retrieved
+ * from reified type parameter.
+ *
+ * @throws [SerializationException] if the given value cannot be serialized to JSON.
+ */
+@ExperimentalApi
+public inline fun <reified T : Any> Ejson.encodeToBsonValue(value: T): BsonValue =
+    encodeToBsonValue(serializersModule.serializer(), value)
+
+
+/**
+ * Deserializes the given [json] element into a value of type [T] using a deserializer retrieved
+ * from reified type parameter.
+ *
+ * @throws [SerializationException] if the given JSON element is not a valid JSON input for the type [T]
+ * @throws [IllegalArgumentException] if the decoded input cannot be represented as a valid instance of type [T]
+ */
+@ExperimentalApi
+public inline fun <reified T : Any> Ejson.decodeFromBsonValue(value: BsonValue): T =
+    decodeFromBsonValue(serializersModule.serializer(), value)
+
+/**
+ * Main entry point to work with EJSON serialization.
+ * [EJSON](https://www.mongodb.com/docs/manual/reference/mongodb-extended-json/)
+ * a string format that extends JSON to support all BSON datatypes.
+ *
+ * A default instance is provided via [Ejson.Default], but if you require an instance with certain
+ * registered serializers or different options you can instantiate it with [Ejson].
+ *
+ * This string format also supports encoding to or from a [BsonValue] with the functions [decodeFromBsonValue]
+ * and [encodeToBsonValue].
+ *
+ * Example of usage:
+ * ```
+ * @Serializable
+ * class DataHolder(val id: Int, val data: String, val bsonValue: BsonValue)
+ *
+ * val ejson = Ejson
+ * val instance = DataHolder(42, "some data", BsonObjectId() }
+ *
+ * // Plain StringFormat usage
+ * val stringOutput: String = ejson.encodeToString(instance)
+ *
+ * // BsonValue serialization
+ * val bsonValue: BsonValue = ejson.encodeToBsonValue(instance)
+ *
+ * // Deserialize from string
+ * val deserialized: DataHolder = ejson.decodeFromString<DataHolder>(stringOutput)
+ *
+ * // Deserialize from BsonValue
+ * val deserializedFromBsonValue: DataHolder = ejson.decodeFromBsonValue<DataHolder>(bsonValue)
+ *
+ *  // Deserialize from string to BsonValue
+ *  val deserializedToBsonValue: BsonValue = Bson(stringOutput)
+ * ```
+ *
+ * It does not support polymorphic serializers yet.
+ */
+@ExperimentalApi
+public sealed class Ejson(
+    public val ignoreUnknownKeys: Boolean,
+    private val json: Json
+) : StringFormat {
+
+    /**
+     * The default instance of [Ejson] with default configuration.
+     */
+    public companion object Default : Ejson(
```

Comment:
Using a `Default` object here seems different than what the Json serializer does. Wouldn't it be better to align with that so they can be used the same way?

Reply:
The Json serializer does use a Default one: [Json.kt:L71](https://github.com/Kotlin/kotlinx.serialization/blob/99496a8cd20fec74d0dabbb2698d48795b405cc5/formats/json/commonMain/src/kotlinx/serialization/json/Json.kt#L71)